### PR TITLE
Collapsing sub-package explosion in Travis config into a script.

### PR DESCRIPTION
**Pull request by dhermes**
This script will be extended a great deal in coming commits.

Note that this doesn't actually make the tests run faster, but it's a minimal change for a PR and I will be sending lots of changes on top of it.

Also, this fixes the current b0rken state of AppVeyor, so that's a plus.

Also, the 2nd commit uses a lesser known feature of `tox`:

- http://tox.readthedocs.io/en/latest/config.html#factors-and-factor-conditional-settings
- http://tox.readthedocs.io/en/latest/config-v2.html

### CODE REVIEW

**Comment by dhermes** [comment]
Green build before I pushed 2nd commit: https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/164164263

**Comment by dhermes** [comment]
@tseaver PTAL

File: scripts/run_unit_tests.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        if not succeeded:
+            failed_packages.append(package)
+
+    if failed_packages:
+        msg_parts = ['Sub-packages failed:']
+        for package in failed_packages:
+            msg_parts.append('- ' + package)
+        msg = '\n'.join(msg_parts)
+        print(msg, file=sys.stderr)
+        sys.exit(len(failed_packages))
```
Comment (daspecster) [inline scripts/run_unit_tests.py]: I get returning something `> 0` here but some codes have specific [meanings](http://tldp.org/LDP/abs/html/exitcodes.html).

Maybe just check if `len(failed_packages) > 0`?

Comment (daspecster) [inline scripts/run_unit_tests.py]: Actually this is probably a better list...
http://www.virtsync.com/c-error-codes-include-errno

Comment (jonparrott) [inline scripts/run_unit_tests.py]: Yeah don't use exit codes to communicate anything other than an `errno`. You could print the number of failed tests before exit.

Comment (dhermes) [inline scripts/run_unit_tests.py]: Got it. Thanks @jonparrott

**Comment by dhermes** [comment]
Merging as-is, putting fix for the error code issue in #2475.

**Comment by dhermes** [comment]
Fingers crossed AppVeyor goes green: https://ci.appveyor.com/project/GoogleCloudPlatform/google-cloud-python/build/1.0.767.master

**Pull request merged**